### PR TITLE
Process groups cannot be created in restricted permission environments

### PR DESCRIPTION
**Pull request by cbrockington**
Setting the process group requires elevated privileges in some environments as it uses the `setgid` capability.

Ideally `go-java-launcher` should not require any additional privileges to run.

### CODE REVIEW

**Comment by gregakinman** [comment]
good to merge @iamdanfox?

**Comment by cbrockington** [comment]
@dansanduleac bump

File: launcher/main/main.go [hunk 1]
```diff
@@ -42,18 +42,24 @@ func ParseMonitorArgs(args []string) (*launchlib.ProcessMonitor, error) {
 	if monitor.PrimaryPID, err = strconv.Atoi(args[0]); err != nil {
 		return nil, errors.Wrapf(err, "error parsing service pid")
 	}
-	if monitor.ProcessGroupPID, err = strconv.Atoi(args[1]); err != nil {
-		return nil, errors.Wrapf(err, "error parsing service group id")
+
+	for _, pidStr := range args[2:] {
```
Comment (cbrockington) [inline launcher/main/main.go]: Probably a bug here, change signature to be `primary string, subs []string`

File: launcher/main/main.go [hunk 2]
```diff
@@ -42,18 +42,24 @@ func ParseMonitorArgs(args []string) (*launchlib.ProcessMonitor, error) {
...
 	}
 	return monitor, nil
 }
 
 func GenerateMonitorArgs(monitor *launchlib.ProcessMonitor) []string {
-	return []string{
-		monitorFlag,
-		strconv.Itoa(monitor.PrimaryPID),
-		strconv.Itoa(monitor.ProcessGroupPID),
+	args := make([]string, 0, len(monitor.SubProcessPIDs)+2)
```
Comment (cbrockington) [inline launcher/main/main.go:57]: spacing

**Review by cbrockington — COMMENTED** [review]
(Really @ferozco's review from my laptop)

Additional things to do:
- [x] Add kill signal propagation from monitor process.
- [x] Testing

File: launcher/main/main.go (hunk 2 above)
Comment (cbrockington) [reply launcher/main/main.go:57]: godel format actually removes the spacing here

**Comment by cbrockington** [comment]
For the integration tests to work this is based on https://github.com/palantir/go-java-launcher/pull/59 with https://github.com/palantir/go-java-launcher/pull/50/commits/14c75e3a0eb263fce883840f2a54f7af92f276c6 being the only additional commit.

File: launcher/main/main.go (hunk 2 above)
Comment (ferozco) [reply launcher/main/main.go:57]: yup seems like gofmt likes it that way 🤷‍♂️

**Review by ferozco — APPROVED** [review]
LGTM

**Pull request merged**